File: LeetCode/Easy/0100-same-tree/0100-same-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public bool IsSameTree(TreeNode p, TreeNode q) {
        if (p == null || q == null)
        {
            return p == q;
        }
        var pStack = new Stack<TreeNode>(100);
        pStack.Push(p);
        var qStack = new Stack<TreeNode>(100);
        qStack.Push(q);



        while (pStack.Count > 0)
        {
            var pNode = pStack.Pop();
            var qNode = qStack.Pop();

            if (pNode.val != qNode.val)
            {
                return false;
            }

            if (pNode.right != null && qNode.right != null)
            {
                pStack.Push(pNode.right);
                qStack.Push(qNode.right);
            }
            else if (pNode.right != null || qNode.right != null)
            {
                return false;
            }

            if (pNode.left != null && qNode.left != null)
            {
                pStack.Push(pNode.left);
                qStack.Push(qNode.left);
            }
            else if (pNode.left != null || qNode.left != null)
            {
                return false;
            }
        }

        return true;        
    }
}

File: LeetCode/Easy/0100-same-tree/0100-same-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        stack: List[TreeNode, TreeNode] = []
        stack.append((p, q))

        while stack:
            p_node, q_node = stack.pop()
            if ((p_node is None) != (q_node is None)):
                return False
            elif p_node is not None and q_node is not None:
                if p_node.val != q_node.val:
                    return False
                stack.append((p_node.right, q_node.right))
                stack.append((p_node.left, q_node.left))
                
        return True